src/Microsoft.Security.Utilities.Benchmarks/SecretMaskerDetectionBenchmarks.cs

```diff
@@ -5,42 +5,85 @@
 
 namespace Microsoft.Security.Utilities.Benchmarks
 {
-    public abstract class SecretMaskerDetectionBenchmarks
+    public enum RegexEngine
     {
-        // The size of randomized data to add as a prefix
-        // for every secret. This is intended to make positive
-        // hit less concentrated in the profiling.
-        private const int SecretPrefixSize = 100 * 1024;
+        DotNet,
+        RE2,
+    }
+
+    public enum Case
+    {
+        // This is close to the worst case scenario for large input. Every
+        // base64 alphabet char is equally likely to appear in the input, making
+        // signature sniffing harder.
+        HardPrefix,
+
+        // This is close to the best case scenario. A char that does not appear
+        // in any identifiable key signature is repeated in the prefix, making
+        // signature sniffing easier.
+        EasyPrefix,
+
+        // This will scan secrets without prefixing them. Testing the performance against
```

Comment:
>Testing [](http://example.com/codeflow?start=58&length=7)

Not a complete sentence. 'This is useful for testing performance against small inputs.' ??